### ISSUE TITLE
Support `serverless run`

### ISSUE THREAD

**Issue by kbrandwijk** [serverless-heaven/serverless-webpack#269]
# This is a Feature Proposal

## Description

For feature proposals:
* What is the use case that should be solved. The more detail you describe this in the easier it is to understand for us.
Add support for `serverless run` by listening to the `before:run:run` hook.

* If there is additional config how would it look
Something similar to this:
https://github.com/serverless-heaven/serverless-webpack/blob/master/index.js#L130-L132

I recently added the same feature to serverless-plugin-typescript (for reference):
https://github.com/graphcool/serverless-plugin-typescript/blob/master/src/index.ts#L31-L37

**Comment by HyperBrain** [comment]
Hi @kbrandwijk . Great idea 🙌 

This would indeed be a great addition. Having the hook implemented would be quite easy. What I'm thinking about is, if a watch option (like the one for sls-offline or sls invoke local) would be possible in the run scenario. I will dig into this a bit - as Docker support was added too recently by the possibility to enable watch polling instead of filesystem events with a switch, that might be possible.

**Comment by kbrandwijk** [comment]
I haven't got the watching part working with `serverless-plugin-typescript` in combination with `serverless run`, so any ideas we can share would benefit both plugins.

**Comment by HyperBrain** [comment]
@kbrandwijk Could you check if my PR works as expected? On my Windows system I get up to where the functions are loaded into the emulator, but then it breaks (Ubuntu shell).
There is a known issue/bug that the event gateway does not work in Windows. I have to check it on my native Ubuntu.

I think I can get the watch mode running - in a similar way as the local invoke does, but just with respawning the run command on a source change. But I have to add and check that on my Ubuntu.

**Comment by kbrandwijk** [comment]
I'm also on Windows. I have that same issue. I just start `sle` manually, then run `sls run` in a different console and it picks up the already running emulator. I'll check the PR version tomorrow.

**Comment by HyperBrain** [comment]
Did not run it physically, but I checked the contents of the emulator storage and they contain the compiled sources which looks ok so far.

**Comment by HyperBrain** [comment]
Added it - now the needed node modules are correctly copied to the emulator.

**Comment by HyperBrain** [comment]
The command seems to work for me now.

I played a bit with getting watch working. I managed to let webpack watch the directory, and it got triggered correctly, but it seems that the run command locks the directory. I have to check if I somehow can get arount the deletion of the service directory for the run command. Then it should work.... stand by, I'm working on it.

```
Serverless: Sources changed.

  Error --------------------------------------------------

  EBUSY: resource busy or locked, rmdir 'C:\Projects\serverless\test\babel-dynam
ically-entries\.webpack\service'
```
